File: apps/ios/Sources/GTRCustomerCore/CatalogChassisAlias.swift
import Foundation

/// Chassis codes in vehicle_master/fixtures without catalog_variants — map to published EPC.
/// Keep in sync with packages/shared catalog-chassis-alias.ts.
public enum CatalogChassisAlias {
    private static let aliases: [String: String] = [
        "D40": "D22",
        "T32": "T31",
    ]

    /// Ordered unique codes to try (requested first, then alias).
    public static func lookupCodes(_ chassis: String) -> [String] {
        let primary = chassis.trimmingCharacters(in: .whitespacesAndNewlines).uppercased()
        guard !primary.isEmpty else { return [] }
        var out = [primary]
        if let alias = aliases[primary]?.trimmingCharacters(in: .whitespacesAndNewlines).uppercased(),
           !alias.isEmpty, alias != primary
        {
            out.append(alias)
        }
        return out
    }
}

File: apps/android-customer/core/rpc/src/main/java/co/zw/nissangtr/customer/rpc/CatalogChassisAlias.kt
package co.zw.nissangtr.customer.rpc

/**
 * Chassis codes present in vehicle_master / fixtures that lack catalog_variants
 * rows on Megazip — map to the published EPC chassis for browse / fitment.
 * Keep in sync with packages/shared catalog-chassis-alias.ts.
 */
object CatalogChassisAlias {
    private val aliases: Map<String, String> = mapOf(
        "D40" to "D22",
        "T32" to "T31",
    )

    /** Ordered unique codes to try (requested first, then alias). */
    fun lookupCodes(chassis: String): List<String> {
        val primary = chassis.trim().uppercase()
        if (primary.isEmpty()) return emptyList()
        val out = mutableListOf(primary)
        aliases[primary]?.trim()?.uppercase()?.takeIf { it.isNotEmpty() && it != primary }?.let {
            out += it
        }
        return out
    }
}
